src/components/challenge-category-grid.tsx

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import { ChallengeFilter, ChallengeFilterCategory } from "../challenge-filters";
+
+import "./challenge-category-grid.scss";
+import ChallengeFilterBox from "./challenge-filter-box";
+import { Challenge } from "../challenges";
+
+interface Props {
+  category: ChallengeFilterCategory;
+  activeFilters: ChallengeFilter[];
+  setActiveFilters: React.Dispatch<React.SetStateAction<ChallengeFilter[]>>;
+  onToggle: (
+    category: ChallengeFilterCategory,
+    selected: boolean,
+    activeFilters: ChallengeFilter[],
+    setActiveFilters: React.Dispatch<React.SetStateAction<ChallengeFilter[]>>,
+  ) => void;
+  changeFilter: (
+    challengeFilter: ChallengeFilter,
+    checked: boolean,
+    activeFilters: ChallengeFilter[],
+    setActiveFilters: React.Dispatch<React.SetStateAction<ChallengeFilter[]>>,
+  ) => void;
+  tier1FilterResults?: Challenge[];
+}
+
+export default function ChallengeCategoryGrid({
+  category,
+  activeFilters,
+  setActiveFilters,
+  onToggle,
+  changeFilter,
+  tier1FilterResults,
+}: Props): React.JSX.Element {
+  const allSelected = category.filters.every(
+    (categoryFilter: ChallengeFilter) => activeFilters.includes(categoryFilter),
+  );
+  const someSelected = category.filters.some(
+    (categoryFilter: ChallengeFilter) => activeFilters.includes(categoryFilter),
+  );
+
+  const handleToggle = (): void => {
+    onToggle(category, !allSelected, activeFilters, setActiveFilters);
+  };
+
+  const onChildToggle = (
+    challengeFilter: ChallengeFilter,
+    selected: boolean,
+  ): void => {
+    changeFilter(challengeFilter, selected, activeFilters, setActiveFilters);
+  };
```

Comment:
We can give more explicit names than `toggle` and `childToggle`. 

We also technically don't need the `onToggle` parameter, since we can just iterate over the `changeFilter` function. I feel like reducing the number of props and functions would be a bit better.

```suggestion
  const handleToggleAll = (): void => {
    for (const filter of category.filters) {
        changeFilter(filter, !allSelected, activeFilters, setActiveFilters);
    }
  };

  const handleToggle = (
    filter: ChallengeFilter,
    selected: boolean,
  ): void => {
    changeFilter(filter, selected, activeFilters, setActiveFilters);
  };
```